### ISSUE TITLE
assign quota FAILED createProjectID

### ISSUE THREAD

**Issue by oliverbaehler** [siderolabs/talos#8652]
## Bug Report

After a Kubelet restart we are having problems with getting the cilium pod up and running, essentially any pod which uses ephermal storage it seems like. The container tries to create an emptDir, however it fails to do so:

```
  Warning  FailedMount  7s (x6 over 32s)  kubelet            MountVolume.SetUp failed for volume "tmp" : assign quota FAILED createProjectID /var/lib/kubelet/pods/5a89cfe1-0738-415e-bc91-83b4952acaa6/volumes/kubernetes.io~empty-dir/tmp 0 failed cannot find available quota ID
```

I have seen this issue, which has recently reported the same behavior (However I don't know if they are also using Talos):
https://github.com/kubernetes/kubernetes/issues/124332. The problem disappears when `machine.features.diskQuotaSupport` is set to `false`

Have you experienced anything like that? Changing the value of the Kubelet FeatureGate `LocalStorageCapacityIsolationFSQuotaMonitoring ` did not change anything about the error.

### Environment

- Talos version: [`talosctl version --nodes <problematic nodes>`]
- Kubernetes version: Server Version: v1.29.2
- Platform:

**Comment by smira** [comment]
Yes, this seems to be a problem in Kubernetes side, but the only thing we can do on Talos side is to disable it by default.
